src/esp/assets/PTexMeshData.cpp

```diff
@@ -614,28 +616,40 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
                         Magnum::GL::MeshIndexType::UnsignedInt);
   }
 
+  // load atlas data and upload them to GPU
+  LOG(INFO) << "loading atlas textures: ";
   for (size_t iMesh = 0; iMesh < renderingBuffers_.size(); ++iMesh) {
-    const std::string rgbFile =
-        atlasFolder_ + "/" + std::to_string(iMesh) + "-color-ptex.rgb";
-    if (!io::exists(rgbFile)) {
-      ASSERT(false, "Can't find " + rgbFile);
-    }
-    LOG(INFO) << "\rLoading atlas " << iMesh + 1 << "/"
-              << renderingBuffers_.size() << "... ";
-    LOG(INFO).flush();
-
-    Cr::Containers::Array<const char, Cr::Utility::Directory::MapDeleter> data =
-        Cr::Utility::Directory::mapRead(rgbFile);
-    const int dim = static_cast<int>(std::sqrt(data.size() / 3));  // square
-    Magnum::ImageView2D image(Magnum::PixelFormat::RGB8UI, {dim, dim}, data);
+    const std::string hdrFile = Corrade::Utility::Directory::join(
+        atlasFolder_, std::to_string(iMesh) + "-color-ptex.hdr");
+
+    CORRADE_ASSERT(io::exists(hdrFile), "Error : Cannot find the .hdr file ", );
+
+    LOG(INFO) << "Loading atlas " << iMesh + 1 << "/"
+              << renderingBuffers_.size() << " from " << hdrFile << ". ";
+
+    Corrade::Containers::Array<const char,
+                               Corrade::Utility::Directory::MapDeleter>
+        data = Corrade::Utility::Directory::mapRead(hdrFile);
```

Comment:
I wonder, does Replica plan to supply the textures in some real file format (with size metadata, compression and such)? With half-floats it could easily be an EXR. Storing (and loading) the data uncompressed is quite wasteful IMHO and this guesswork with calculating image size is error-prone.

Reply:
I see. To answer this question, we will have to ask our colleagues at FRL who wrote the replicaSDK. 
This code is simply following that.

Reply:
Sure -- I'm not saying this code is to blame ;) It just felt strange to me, so I figured you would know the reason why they did it like this.